[TypeChecker] `<unknown>` diagnostic location regarding `Codable` derived conformances

HassanElDesouky opened the pull request:
<!-- What's in this pull request? -->
Fixing `<unknown>` diagnostic location regarding `Codable` derived conformances.

cc @LucianoPAlmeida, @CodaFi, @xedin  :) 

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
rdar://problem/59655704
Resolves [SR-12248](https://bugs.swift.org/browse/SR-12248).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

modocache committed:
Document what a "subcommand" is

There are a few references to "subcommands" in the repository, but no
explanation of what they are. Beef up the docblock for
`shouldRunAsSubcommand()` to make things a little clearer.
Specifically, I'd like to make the difference between driver jobs and
driver subcomnmands clearer.

In addition, remove a double negative comment, "If we are not run as
'swift', don't do anything special", and replace it with something a
little clearer (in my opinion).

kubamracek committed:
Add a test for a compiler crasher with -O -whole-module-optimization -sanitize=address

rockbruno committed:
Allow unavailable decls to be walked

owenv committed:
[Diagnostics] Document the verifier syntax for educational notes

rockbruno committed:
Update index test response to cover unavailable calls

own2pwn committed:
Add steps to enable -Ounchecked compiler flag

rintaro committed:
[CodeCompletion] Use AnyFunctionType::Param::getParameterType()

instead of getOldType() in KeyPath as funtions completion.
Also, don't think keypath's are convertible to functions that receives
variadic parameters.

theblixguy committed:
[Typechecker] Wrapped properties with observers should not have ReadWriteImplKind::Modify

theblixguy committed:
[Typechecker] Supress _modify synthesis if the wrapped property has observers

theblixguy committed:
[Test] Add an executable test

hamishknight committed:
[CS] Visit all fixed bindings for constraint re-activation

Start visiting transitive fixed bindings for type
variables, and stop visiting adjacencies for
`gatherConstraint`'s `AllMentions` mode.

This improves performance and fixes a correctness
issue with the old implementation where we could
fail to re-activate a coercion constraint, and
then let invalid code get past Sema, causing
either miscompiles or crashes later down the
pipeline.

Unfortunately this change requires us to
temporarily drop the non-ephemeral fix for a couple
of fairly obscure cases where the overload hasn't
yet been resolved. The logic was previously relying
on stale adjacency state in order to re-activate
the fix when the overload is bound, but it's not
connected on the constraint graph. We need to find
a way to connect constraints to unresolved
overloads they depend on.

Resolves SR-12369.

hamishknight committed:
[CS] Account for type variables when matching metatypes

Previously we could prematurely attempt to perform
a bind of class metatypes without checking for
subtyping. Tweak the logic to not perform a bind
if we can't prove that we're dealing with non-class
types.

hamishknight committed:
Revert "[Constraint graph] Reinstate the adjacencies of constraint graph nodes."

This reverts commit cf1732cce21b7e7f68c5c399ed9e1585d1919c2c.
With `gatherConstraints` no longer visiting
adjacencies, we no longer need to maintain the
adjacency info.

hamishknight committed:
[CS] Preserve compatibility for collection coercions

Previously we could allow some invalid coercions to
sneak past Sema. In most cases these would either
cause crashes later down the pipeline or
miscompiles. However, for coercions between
collections, we emitted somewhat reasonable code
that performed a force cast.

This commit aims to preserve compatibility with
those collection coercions that previously
compiled, and emits a warning telling the user to
use either 'as?' or 'as!' instead.

hamishknight committed:
[CS] Assert that we don't end up with unsolved constraints

Make sure we don't end up in a situation where we
have unsolved constraints left over and consider
the system fully solved.

This requires tweaking the type matching code for
dependent members such that a concrete base is
considered a failure rather than being left
unsolved. This should only happen when not in
diagnostic mode, as otherwise we use a hole.

owenv committed:
[DiagnosticVerifier] Add support for asserting presence of edu notes

owenv committed:
[Diagnostics] Expand test coverage of existing edu notes

milseman committed:
[gardening] Delete Trailing Whitespace

milseman committed:
[string] _HasContiguousBytes -> withContiguousStorageIfAvailable

Switch String(decoding:as) and other entry points to call
withContiguousStorageIfAvailable rather than use _HasContiguousBytes.

Commit:
Convert a clang::Module to non-const to account for a clang API change

An upstream clang changed ASTSourceDescriptor to not have a const
Module pointer. const_cast here to make this agree.

(cherry picked from commit a49bceeedf2875c6542812b0dc03ae7e382246e1)

hamishknight committed:
[CS] NFC: Move depthFirstSearch up slightly

HassanElDesouky committed:
[SR-12248] Fix <unknown> location from synthesized decl

HassanElDesouky committed:
[SR-12248] Add tests

LucianoPAlmeida commented on test/decl/protocol/special/coding/struct_codable_simple.swift:
```diff
@@ -29,3 +29,21 @@ let _ = SimpleStruct.encode(to:)
+struct SR_12248_1: Codable { // expected-error {{type 'SR_12248_1' does not conform to protocol 'Encodable'}}
+  var x: Int // expected-note {{'x' previously declared here}}
+  var x: Int // expected-error {{invalid redeclaration of 'x'}}
+  // expected-note@-1 {{cannot automatically synthesize 'Encodable' because '<<error type>>' does not conform to 'Encodable'}}
```
Not sure if this is a requirement for the SR, but I think this shouldn't be noted when the synthesize type `hasError()`, in fact, the missing `Encodable` conformance diagnostic itself shouldn't be diagnosed in this case.
WDYT?

HassanElDesouky replied:
Honestly, I also thought of the same thing. I don’t think this needs to be tested. However, as a newbie I decided to add a lot of tests 😅

LucianoPAlmeida replied:
You could take a look at `DerivedConformance::deriveEncodable`, note that in 

https://github.com/apple/swift/blob/426583523e063be1ed75027044f56873b7b73183/lib/Sema/TypeCheckDeclPrimary.cpp#L721 

The member is flagged as invalid, so on `DerivedConformance::deriveEncodable` and decodable you could check for any nominal member and skip the conformance diagnostic if any of the members are invalid :)

CodaFi replied:
The `ErrorType`s are unfortunate, but it's not enough to block the PR.

milseman committed:
[string] Outline cold path from initializer

Outline the cold, non-contiguous UTF-8 path from String(decoding:as:),
saving ~40 bytes (33%) of code size (x86_64 and arm64) from every call
site where the contiguity check cannot be constant folded away.

slavapestov committed:
IRGen: Type reconstruction supports opaque result types now

Commit:
[Parse] Don't drop the EnumCaseDecl when it has a trailing comma.

This meant we weren't producing sema diagnostics for the case, and it didn’t
get full syntactic/semantic highlighting or indentation.

enum CasesWithMissingElement {
  case a(Int, String),
  case b(Int, String),
}

Resolves rdar://problem/61476844

Commit:
[IDE][Refactoring] Handle 'callAsFunction' specially in syntactic rename.

This change makes us treat it exactly as we do 'init'. We don't allow renaming the base name,
and don't fail if the basename doesn't match for calls.

Also:
  - explicit init calls/references like `MyType.init(42)` are now reported with
    'init' as a keywordBase range, rather than nothing.
  - cursor info no longer reports rename as available on init/callAsFunction
    calls without arguments, as there's nothing to rename in that case.
  - Improved detection of when a referenced function is a call (rather than
    reference) across syntactic rename, cursor-info, and indexing.

Resolves rdar://problem/60340429

compnerd committed:
test: repair the AutoDiff test on Windows

`-lm` is not portable, it is only needed on certain platforms (e.g.
Linux, not BSD).  Remove the explicit link.

Avoid using a static library as that is not supported on all platforms
(e.g. Windows).  Use the helper macros to generate the library.

Repairs the Windows tests.

dan-zheng committed:
Merge pull request #30875 from apple/autodiff-upstream-stdlib-differentiation

[AutoDiff upstream] Add stdlib `Differentiable` conformances and derivatives.

swift-ci committed:
Merge pull request #30913 from atrick/fix-assert-termoper

dan-zheng committed:
Merge pull request #30912 from dan-zheng/fix-synthesized-file-unit

Fix SynthesizedFileUnit serialization and TBDGen issues.

dan-zheng committed:
[AutoDiff] Remove obsolete TODO comments.

dan-zheng committed:
[Gardening] Remove unnecessary forward declaration.

dan-zheng committed:
[test] Add `config.target_run_simple_swiftgyb_parameterized`.

Support `%target-run-simple-swiftgyb(...)` taking compiler arguments.
Copies existing support for `%target-run-simple-swift(...)`.

dan-zheng committed:
[AutoDiff] Enable forward-mode differentiation tests.

Resolves TF-1237 TODO comments.

swift-ci committed:
Merge pull request #30917 from dan-zheng/autodiff-gardening

tbkka committed:
Float16 optimal formatting (#30862)

Extend SwiftDtoa to provide optimal formatting for Float16 and use that for `Float16.description` and `Float16.debugDescription`.

Notes on signaling NaNs: LLVM's Float16 support passes Float16s on x86
by legalizing to Float32.  This works well for most purposes but incidentally
loses the signaling marker from any NaN (because it's a conversion as far
as the hardware is concerned), with a side effect that the print code never
actually sees a true sNaN.  This is similar to what happens with Float and
Double on i386 backends.  The earlier code here tried to detect sNaN in a
different way, but that approach isn't guaranteed to work so we decided to
make this code use the correct detection logic -- sNaN printing will just be
broken until we can get a better argument passing convention.

Resolves rdar://61414101

owenv committed:
Merge pull request #30879 from owenv/cleanup-case-typechecking

[Sema] Eliminate duplication in CaseStmt typechecking for switch and do-catch

compnerd committed:
Merge pull request #30911 from compnerd/long-paths

test: support long paths in `rewrite-module-triples`

compnerd committed:
test: lit still requires python2, explicitly use that

This updates the invocation to use `Python2::Interpreter` rather than
relying on the custom `PYTHON_EXECUTABLE`.  This is much easier to
identify where we are falling back to the python2 code paths and where
we can start moving towards Python3.

Commit:
[AutoDiff upstream] handle differentiable_function in DiagnoseInvalidEscapingCaptures (#30909)

brettkoonce committed:
rm minor warning

dan-zheng committed:
Merge pull request #30918 from dan-zheng/autodiff-forward-mode

[AutoDiff] Enable forward-mode differentiation tests.

compnerd committed:
Merge pull request #30921 from compnerd/python2

test: lit still requires python2, explicitly use that

DougGregor committed:
[Constraint solver] Migrate for-each statement checking into SolutionApplicationTarget.

Pull the entirety of type checking for for-each statement headers (i.e., not the
body) into the constraint system, using the normal SolutionApplicationTarget-based
constraint generation and application facilities. Most of this was already handled
in the constraint solver (although the `where` filtering condition was not), so
this is a smaller change than it looks like.

dan-zheng committed:
[AutoDiff upstream] Add `Differentiable.zeroTangentVector`.

Add `Differentiable.zeroTangentVectorInitializer` protocol requirement and
`Differentiable.zeroTangentVector` default implementation.

DougGregor committed:
[Constraint solver] Remove ExprTypeCheckListener.

The last client of this listener-based interface has been refactored,
so remove it.

dan-zheng committed:
[AutoDiff upstream] Add `AnyDerivative`.

Add `AnyDerivative`, a type-erased wrapper for `Differentiable.TangentVector`
associated type implementations.

compnerd committed:
test: repair the Android CI after #30875

The added test was not correctly indicated as a running test.

rintaro committed:
[SourceKit] Avoid one memcpy on potentially big sourcekit response data

A kind indicator is needed before the actual data when custom data are
storeed as xpc dictionary values. Instead of prepending the kind bit by
copying data to another buffer, let the data producers include it in the
created data.

compnerd committed:
runtime: fix a UBSAN issue

This was identified by UBSAN: signed-integer-overflow.  Explicitly mark
the value as unsigned to ensure that the value does not overflow.  There
is a second instance of a raw literal, however, because it is a `*=` the
value is implicitly understood to be unsigned.

xedin committed:
[Diagnostics] Replace `getChoiceFor` with `getCalleeOverloadChoiceIfAvailable`

New name is more consistent with existing `getOverloadChoiceIfAvailable`
and allows us to clean up a couple of places where `getCalleeLocator` was
used directly before.

Commit:
Merge pull request #30894 from nathawes/avoid-multiline-string-indent

[SourceKit/CodeFormat] Don't modify multiline string indentation regardless of it being column-aligned or not.

slavapestov committed:
AST: Fix opaque type mangling used by type reconstruction

Make sure we mangle opaque types using the same settings as the
debugger mangling (with OptimizeProtocolNames = false) to ensure
that we can reconstruct those names again.

slavapestov committed:
Serialization: Implement LocalDeclTableInfo::GetExternalKey()

This allows iterating over serialized decl tables, which is useful
for debugging.

milseman committed:
[string] Restore _HasContiguousBytes for untyped storage

UnsafeRawBufferPointer cannot implement
withContiguousStorageIfAvailable because doing so would potentially
create a typed pointer from untyped data.

dan-zheng committed:
Merge pull request #30926 from dan-zheng/autodiff-stdlib

[AutoDiff upstream] Upstream `_Differentiation` APIs.

compnerd committed:
Merge pull request #30925 from compnerd/30875

test: repair the Android CI after #30875

Commit:
[AutoDiff upstream] DifferentiationUnittest and some e2e tests (#30915)

Adds 2 simple e2e tests and some lit subsitutions and unittest libraries
necessary to support them.

gottesmm committed:
[polymorphic-builtins] Enable test during optimize builds.

The confusion here was around a test that was testing the behavior when one
/does not/ guard a polymorphic builtin with a _isConcrete. Specifically, without
_isConcrete, one gets weird behavior where in debug we crash, but in release we
may not due to further inlining/specialization. With _isConcrete, we never
crashand get the appropriate behavior.

This commit works by changing the current run of the polymorphic builtin test to
force -Onone and adds an extra -O run that way we validate both behaviors.

compnerd committed:
Merge pull request #30927 from compnerd/undefined-is-the-only-desired-behaviour

runtime: fix a UBSAN issue

CodaFi committed:
Merge pull request #30795 from davidungar/typo-fix

[NFC] Fix a typo in a comment

omochi committed:
[Diagnostics] Improve {{none}} fix-it verifier (#30791)

* [Diagnostics] Improve {{none}} fix-it verifier

* split two conditions

* define "none" constant

* support plural

* use Twine and add comment for replacement range

* check if {{none}} is at the end

* use noneMarkerStartLoc

* update second {{none}} error message

Co-Authored-By: Owen Voorhees <owenvoorhees@gmail.com>

* update test case for second {{none}}

* fix test case for new {{none}} check

* Use named struct

* set const

Co-authored-by: Owen Voorhees <owenvoorhees@gmail.com>

xedin committed:
[Diagnostics] Remove unsafe `getRestrictionFor` method

hborla committed:
[Property Wrappers] Introduce a new Expr node for the property wrapper
wrapped value placeholder in an init(wrappedValue:) call that was previously
injected as an OpaqueValueExpr. This commit also restores the old design of
OpaqueValueExpr.

CodaFi committed:
[Gardening] Apply Review Feedback

hborla committed:
[SILGen] Remove an unnecessary scope in emitImplicitValueConstructor

CodaFi committed:
Merge pull request #30881 from owenv/se-0276-changelog

[Changelog] Add SE-0276 to the changelog

theblixguy committed:
[Typechecker] Add an hasObservers() convenience method and simplify some code

slavapestov committed:
Sema: Force property wrappers in EmittedMembersRequest::evaluate()

Fixes <rdar://problem/61229365>.

slavapestov committed:
Sema: Force property wrappers in EmittedMembersRequest::evaluate()

Fixes <rdar://problem/61229365>.

artemcm committed:
[Sema] Diagnose use of implementation-only property wrappers
We already ban all structs from declaring storage that comes from implementation-only imports. Until now we missed property wrappers, they were just dropped in deserialization.

Resolves rdar://problem/59403617

compnerd committed:
Platform: alias math constants on Windows

The non-C/C++ standard defined math constants require
`-Xcc -D_USE_MATH_DEFINES` when using the `ucrt` module.  Instead,
provide them through the wrapper `MSVCRT` module as aliases so that the
user does not need to be aware of this in practice.

Repairs the Windows build of `_Differentiation`.

gottesmm committed:
[debugging-the-compiler] Add to bisecting optimization passes a short example of how to use llvm's bisect tool to autobisect the optimizer.

compnerd committed:
AutoDiff: Android, Windows does not support FP80

Correct the condition for FP80.  This should have no real impact on
Android as the tests are not currently executed.

compnerd committed:
AutoDiff: disable the test cases for FP80

The FP80 tests should not be built on Android or Windows.  This fixes
the Windows run of the test.

compnerd committed:
AutoDiff: repair the cross-module-differentiation on Windows

Make the cross-module differentiation test work on Windows by converting
to a shared library to run the test.

compnerd committed:
AutoDiff: repair the test on Windows

The name of the module is encoded into the type name.  The module name
is not `null` on Windows, which would fail to match the name.  It is
possible to make the test less overly-constricted, but setting the
module name to null is simpler and avoids more regex matching.

compnerd committed:
vim: add `differentiability_witness` to SIL syntax

Add the keyword from the experimental auto-differentiation functionality.

dan-zheng committed:
[AutoDiff] Add `AnyDifferentiable`. (#30928)

Add `AnyDifferentiable`, a type-erased wrapper for `Differentiable`-conforming types.

atrick committed:
Fix an assert in OwnershipUtils.

checked_cast_br may take an additional operands for the source and
target types. I'm sure the compiler forgets to check this in many
places. In this case, it was just a harmless assert.

Fixes <rdar://61122253> Assertion failed:
(termInst->getNumOperands() == 1 && "Transformation terminators should only have single operands")

DougGregor committed:
Revert "[PrintAsObjC] Don't include the module for empty extensions"

This reverts commit 8f01b2c8f8ccb2b2d5337029df1aa85ed994242c.

xedin committed:
Merge pull request #30932 from xedin/failure-diagnostic-gardening

[Diagnostics] A couple of adjustments to `FailureDiagnostic`

swift-ci committed:
Merge pull request #30943 from atrick/fix-assert-termoper

dan-zheng committed:
[AutoDiff] Support direct `init` reference differentiation.

Support `@differentiable` function conversion for `init` references, in
addition to `func` references and literal closures. Minor usability improvement.

Resolves SR-12562.

dan-zheng committed:
[AutoDiff] Fix `@derivative` attribute type-checking crash. (#30936)

Fix `@derivative` attribute type-checking crash, so far reproducible only via
`-parse-stdlib`.

The crash occurs because it is not sufficient for type-checking to check for
`Differentiable` conformances. We must also check for invalid `TangentVector`
associated types.

Resolves SR-12559.

eeckstein committed:
EscapeAnalysis: remove an unused parameter from canOptimizeArrayUninitializedCall

NFC

eeckstein committed:
EscapeAnalysis: don't compute the connection graph for very large functions

For functions which results in > 10000 nodes, just bail and don't compute the connection graph.
The node merging algorithm is quadratic and can result in significant compile times for very large functions.

rdar://problem/56268570

eeckstein committed:
AliasAnalysis: speed up canApplyDecrementRefCount() for large functions.

This is a workaround for some quadratic complexity in ARCSequenceOpt which calls canApplyDecrementRefCount very frequently.

rdar://problem/56268570

eeckstein committed:
SILCombine: limit the worklist size for instruction canonicalization

To be precise: don't add instruction uses to the worklist if it already has more than 10000 elements.
This avoids quadratic behavior for very large functions.

rdar://problem/56268570

eeckstein committed:
Add an option to completely disable SIL verification.

This is useful to disable SIL verification in an assert build of the compiler.

owenv committed:
[DiagnosticVerifier] Fix ASAN issue where Twine was stored to a temporary

DougGregor committed:
Merge pull request #30924 from DougGregor/for-each-solution-application-target

[Constraint solver] Migrate for-each statement checking into SolutionApplicationTarget

compnerd committed:
Merge pull request #30940 from compnerd/magic-numbers-are-good

Platform: alias math constants on Windows

CodaFi committed:
Merge pull request #30935 from CodaFi/maybe-driver

[driver][Shepherd] Document what a "subcommand" is (NFC)

eeckstein committed:
Support try_apply and yield in swift::replaceBranchTarget

eeckstein committed:
SimplifyCFG: fix a compile time problem with block merging

When merging many blocks to a single block (in the wrong order), instructions are getting moved over and over again.
This is quadratic and can result in very long compile times for large functions.
To fix this, always move the instruction to smaller block to the larger block.

rdar://problem/56268570

CodaFi committed:
Merge pull request #30735 from owenv/verifier-edu-notes

[DiagnosticVerifier] Support asserting edu notes are emitted

CodaFi committed:
Merge pull request #30934 from CodaFi/unchecked-opt-imism

[Gardening] Add Missing Instructions For Setting Optimization Levels in Xcode

theblixguy committed:
[Changelog] Add SE-0268 to the changelog, under Swift 5.3 (#30903)

owenv committed:
Merge pull request #30947 from owenv/verifier-asan-fix

[DiagnosticVerifier] Fix ASAN issue in the verifier

rintaro committed:
[CodeCompletion] Add pretty stacktrace for completion operation

CodaFi committed:
Merge pull request #30922 from brettkoonce/rm-benchmark-warning

[benchmark] rm minor warning

gottesmm committed:
Merge pull request #30930 from gottesmm/pr-e789a43b68653b4658b5c65d1d836ee36a1074e2

[polymorphic-builtins] Enable test during optimize builds.

hborla committed:
[Property Wrappers] Don't continue on to synthesizing and type checking a
property wrapper backing init if the property type doesn't match the
wrapped value type.

compnerd committed:
Merge pull request #30949 from compnerd/fp80-android

AutoDiff: Android, Windows does not support FP80

compnerd committed:
Merge pull request #30952 from compnerd/autodiff-vs-differential-programming

AutoDiff: repair the test on Windows

eeckstein committed:
tests: fix the timeout.py script and move it to a more central place

A string->float conversion for the command line argument was missing

eeckstein committed:
tests: add a compile-time test for large string arrays.

rdar://problem/56268570

milseman committed:
[string] Move wCSIA check higher than _HasContiguousBytes

milseman committed:
[string] Convert IR tests to SIL tests

slavapestov committed:
IRGen: Fix simple_display(IRGenDescriptor)

slavapestov committed:
Merge pull request #30937 from slavapestov/fix-property-wrapper-vtable-layout-with-subclass

Sema: Force property wrappers in EmittedMembersRequest::evaluate()

dan-zheng committed:
[AutoDiff upstream] Add `Differentiable.withDerivative(_:)`. (#30945)

Add `Differentiable.withDerivative(_:)`, a "derivative surgery" API.

`Differentiable.withDerivative(_:)` is an identity function returning `self`.
It takes a closure and applies it to the derivative of the return value, in
contexts where the return value is differentiated with respect to.

dan-zheng committed:
[AutoDiff] Support direct `init` reference differentiation. (#30946)

Support `@differentiable` function conversion for `init` references, in
addition to `func` references and literal closures. Minor usability improvement.

Resolves SR-12562.

hamishknight committed:
[CS] Visit all fixed bindings for constraint re-activation (#30886)

[CS] Visit all fixed bindings for constraint re-activation

xedin committed:
[Diagnostics] Add a `getLoc` method to `FailureDiagnostic`

rintaro committed:
Merge pull request #30954 from rintaro/ide-completion-prettystacktrace

[CodeCompletion] Add pretty stacktrace for completion operation

rintaro committed:
Merge pull request #30929 from rintaro/sourcekit-setcustombuffer-avoidmemcopy

[SourceKit] Avoid one memcpy on potentially big sourcekit response data

theblixguy committed:
[Typechecker] Simplify StorageImplInfo code in finishPropertyWrapperImplInfo and check for AccessorKind in synthesizeCoroutineAccessorBody

theblixguy committed:
[Test] Add a SILGen test to make sure the observers are correctly called

slavapestov committed:
Add test cases for initializer replacement to test/Interpreter/dynamic_replacement.swift

texasmichelle committed:
Add .gyb extension to filename

slavapestov committed:
Fix formatting nits in test/Interpreter/dynamic_replacement.swift

compnerd committed:
Merge pull request #30951 from compnerd/cross-module-differentiation-win

AutoDiff: repair the cross-module-differentiation on Windows

nkcsgexi committed:
AST: making `export: true` in @_specialized attribute a no-operation

The client code doesn't actually call into these specialized functions even
though they have public linkage. This could lead to TBD verification failure
shown in rdar://44777994.

This patch also warns users' codebase when `export: true` is specified.

xedin committed:
[Diagnostics] Allow failure diagnostic implementation to extend `getAnchor`

Some of the diagnostics have special rules about anchor location,
let's make `getAnchor` virtual and allow sub-class to override
default implementation.

compnerd committed:
Merge pull request #30950 from compnerd/no-fp80

AutoDiff: disable the test cases for FP80

artemcm committed:
Merge pull request #30939 from artemcm/ImplOnlyPropertyWrappers

[Sema] Diagnose use of implementation-only property wrappers

fredriss committed:
Merge pull request #30953 from adrian-prantl/deconstify-master

Convert a clang::Module to non-const to account for a clang API change

slavapestov committed:
IRGen: Enable dynamic replacement with library evolution

It looks like the only thing that fails is the linkage computation
for the dynamic replacement key of class methods. Even though
methods have hidden linkage to prevent them from being directly
referenced from outside a resilient module, we need to ensure
the dynamic replacement key is visible.

Fixes <rdar://problem/58457716>.

benlangmuir committed:
[build-script] Add sanitizer support to indexstore-db product

Support the usual `--enable-*san options`, but also add a
`--test-indexstore-db-santitize-all` that runs the tests once for each
sanitizer. Sanitizing just indexstore-db with a regular toolchain should
be much faster than using sanitized compilers.

milseman committed:
Merge pull request #30729 from milseman/merge_me_if_available

[string] String(decoding:as:) fast path for withContiguousStorageIfAvailable

xedin committed:
[Diagnostics] Use new `getLoc` and `getSourceRange` methods

Split `emitDiagnostic` into `emitDiagnostic` and `emitDiagnosticAt`.

Refactor existing diagnostics to use new methods and avoid passing
location when possible.

Commit:
Tolerate non-empty braces in braces

compnerd committed:
Merge pull request #30907 from compnerd/autodiff-fix

test: repair the AutoDiff test on Windows

Commit:
Test nonempty-brace-in-brace

nkcsgexi committed:
Merge pull request #30962 from nkcsgexi/44777994

AST: making `export: true` in @_specialized attribute a no-operation

Commit:
Merge pull request #30870 from nathawes/mutiple-catch-pattern-indentation

[SourceKit/CodeFormat] Column-align multiple patterns in catch clauses

rintaro committed:
Merge pull request #29854 from rintaro/ide-completion-keypathfunc-paramty

[CodeCompletion] Use AnyFunctionType::Param::getParameterType()

swift-ci committed:
Merge pull request #30899 from nathawes/improve-enum-case-trailing-comma-recovery

slavapestov committed:
Merge pull request #30861 from slavapestov/outdated-fixme

IRGen: Type reconstruction supports opaque result types now

eeckstein committed:
Merge pull request #30957 from eeckstein/fix-large-string-arrays

Several compile time fixes related to large string arrays.

slavapestov committed:
Merge pull request #30965 from slavapestov/enable-dynamic-replacement-vs-library-evolution

Enable dynamic replacement with library evolution

Commit:
[DiagnosticVerifier] Explicit convert a StringRef into a std::string.

The newer llvm StringRef library has removed the implicit StringRef to
std::string conversion.  (See: llvm/llvm-project@adcd026)

This patch also uses a StringRef to compare another StringRef
eliminating the need to perform a StringRef to std::string conversion.
I am concerned that StringRef's are being stored in
ExpectedEducationalNotes, but as long as the StringRef does not out live
the definition this is totally cool then.

Commit:
Merge pull request #30968 from davidungar/rdar-61461187-tolerate-nonempty-brace-in-brace

[ASTScope lookup] Tolerate non-empty braces in braces

compnerd committed:
Merge pull request #30885 from compnerd/android-fp80

stdlib: android does not support fp80

compnerd committed:
test: speculative fix after #30965

apple/swift#30965 broke the Windows test suite, this speculatively fixes
it.

compnerd committed:
build: remove Swift search paths for host

Remove the Swift library search path for the host paths which are C/C++
only.

compnerd committed:
build: prune some unused options for `_add_swift_host_executable_single`

Remove some unused parameters for this function.  This simplification
will make it easier to merge `_add_swift_host_executable_single` into
`add_swift_host_executable`.

compnerd committed:
Merge pull request #30974 from compnerd/vim

vim: add `differentiability_witness` to SIL syntax

compnerd committed:
build: duplicate and rename the `_add_variant_*` functions

This allows us to start splitting up the swift and C/C++ specific paths.

compnerd committed:
Merge pull request #30975 from compnerd/30965

test: speculative fix after #30965

owenv committed:
Merge pull request #30973 from enferex/explicit-stringref-to-string

[DiagnosticVerifier] Explicitly convert a StringRef into a std::string.

compnerd committed:
Merge pull request #30961 from texasmichelle/change_filename

Add .gyb extension to filename

compnerd committed:
Merge pull request #30976 from compnerd/unused-paths

build: remove Swift search paths for host

compnerd committed:
Merge pull request #30977 from compnerd/unused

build: prune some unused options for `_add_swift_host_executable_single`

omochi committed:
[run-test] add `--build-jobs` option

drodriguez committed:
[android] Add check for __aarch64__ besides __arm__ and __arm64__.

The spelling __arm64__ seems to be an Apple-only thing, which is not
provided when targetting Android, and probably other AArch64 Linuxes.

nkcsgexi committed:
ABI-checker: diagnose missing available attributes for members of extensions instead of the extensions themselves

rdar://61644469

drodriguez committed:
Merge pull request #30982 from drodriguez/android-cfuncs-parse-check-aarch64

[android] Add check for __aarch64__ besides __arm__ and __arm64__.

dabrahams committed:
Add -sil-cross-module-serialize-all option. (#30816)

Thanks, Michael!

nkcsgexi committed:
Merge pull request #30983 from nkcsgexi/61644469

ABI-checker: diagnose missing available attributes for members of extensions instead of the extensions themselves

RLovelett committed:
test: Add parens to print for Python3 compatibility

CodaFi committed:
Merge pull request #30082 from 3405691582/OpenBSD_Port_Build

Build system recognizes OpenBSD.

CodaFi committed:
Merge pull request #15108 from darquro/fast-path-ad-hoc-printing-of-strings

[stdlib][SR-7136] Fast-path ad-hoc printing of Strings

CodaFi committed:
Merge pull request #30981 from omochi/run-test-jobs

[run-test] add `--build-jobs` option

compnerd committed:
WinSDK: add some constants for the SDK overlay

Unfortunately, the types are not imported through the clang importer.
Add them to the overlay for use in applications.

slavapestov committed:
Merge pull request #29931 from theblixguy/fix/SR-12178

[Typechecker] Fix _modify for wrapped properties with observers

compnerd committed:
Platform: add ShellCore to WinSDK modulemap

Add the ShellCore module to the Windows SDK modulemap.

eeckstein committed:
tests: The large_string_array test only works for 64 bits.

swift-ci committed:
Merge pull request #30991 from eeckstein/fix_string_array_test

compnerd committed:
Merge pull request #30990 from compnerd/shellcore

Platform: add ShellCore to WinSDK modulemap

compnerd committed:
Merge pull request #30988 from compnerd/constantly-improving

WinSDK: add some constants for the SDK overlay

compnerd committed:
build: explicitly execute `line-directive-tool` with Python3

The `line-directive-tool` is Python3 compliant.  Use the Python3
interpreter to execute this tool.

Commit:
[AutoDiff upstream] add more differentiation tests (#30933)

Commit:
Merge pull request #30989 from CodaFi/of-no-avail

[Index][SR-9567] Allow unavailable decls to be walked

shahmishal committed:
Merge pull request #30698 from AustinConlon/patch-1

[Gardening] Change Xcode link

Commit:
Merge pull request #30964 from nathawes/syntactic-rename-for-callAsFunction

[IDE][Refactoring] Handle 'callAsFunction' specially in syntactic rename

rintaro committed:
[CodeCompletion][NFC] Fix typo

Commit:
Merge pull request #30891 from nathawes/indent-enum-case-decl

[SourceKit/CodeFormat] Column-align enum element decls and the items in their parameter lists.

compnerd committed:
stdlib: android does not support FP80

Correct the maximal floating point width.  Although technically, Android
could support FP128 on AArch64, Swift does not currently support FP128.
For now, ensure that we use FP64 on Android.

CodaFi committed:
Merge pull request #30609 from gribozavr/sil-rst

Removed a description of curried functions

bitjammer committed:
[SymbolGraph] Ignore some Self requirements

To ease the burden on the client, ignore some generic requirements involving
Self. For example, `Self: P` where we already know that `Self` conforms to `P`.

An example case:

```
public struct S: Equatable {
  public static func ==(lhs: S, rhs: S) -> Bool { ... }
}
```

`!=` is defined in terms of `Self` and the default implementation has a `Self:
Equatable`. For the purposes of documentation, it's not necessary to specify
that again on the page of documentation for `!=`.

rdar://60963924

swift-ci committed:
Merge pull request #30548 from AnthonyLatsis/sr-7855

drexin committed:
Merge pull request #30978 from compnerd/variants

build: duplicate and rename the `_add_variant_*` functions

DougGregor committed:
Revert "[PrintAsObjC] Test that we don't leak extended implementationOnly types"

This reverts commit 4e8867e02123079020d1ac77e00c0c34e0bedf16.

CodaFi committed:
Tighten-up a Foundation Deserialization Test

CocoaError.Code.fileReadUnknown conflicts with the CHECK-NOT line here.
We should be checking more specifically for UnknownBlock and UnknownCode
anyways.

rdar://53284293

compnerd committed:
Merge pull request #30997 from compnerd/svelte-fp

stdlib: android does not support FP80

bitjammer committed:
[SymbolGraph] Look at inherited protocols for default implementations

rdar://61178480

bitjammer committed:
Merge pull request #30998 from bitjammer/acgarland/rdar-60963924-ignore-some-self-reqs

[SymbolGraph] Ignore some Self requirements

benlangmuir committed:
Fix macOS indexstore-db preset

rintaro committed:
Merge pull request #30996 from rintaro/ide-completion-performcachedoperaiton

[CodeCompletion][NFC] Fix typo

zoecarver committed:
[SIL] [Parser] Move ParserSIL into SIL library.

Move the source files from ParserSIL into the SIL library and remove the
ParserSIL library. ParsersSIL doesn't need to be its own library and this change will
simplify our builds.

compnerd committed:
Merge pull request #30992 from compnerd/the-impossible-future

build: explicitly execute `line-directive-tool` with Python3

hborla committed:
[Property Wrappers] Use the outermost wrapper attribute type as the
contextual type for property wrapper initialization

lorentey committed:
Merge remote-tracking branch 'origin/master' into fortify-benchmarks

lorentey committed:
[benchmark] Fix thinko

valeriyvan committed:
Removes redundant buffer zeroing in CGColorSpace.swift by using `init(unsafeUninitializedCapacity:initializingWith:) (#30134)

* Removes redundant buffer zeroing

* Fix compile error

swift-ci committed:
Merge pull request #31003 from zoecarver/sil-parser/move-lib

compnerd committed:
build: explicitly pass the path to python2,python3

Explicitly pass the path to the python interpreter on macOS by using
`xcrun` to find the tools.

hborla committed:
Merge pull request #31002 from hborla/property-wrapper-contextual-type

[Property Wrappers] Use the outermost wrapper attribute type as the contextual type for property wrapper initialization

DougGregor committed:
Merge pull request #30944 from DougGregor/import-module-for-empty-extensions

Revert "[PrintAsObjC] Don't include the module for empty extensions"

CodaFi committed:
Merge pull request #30999 from CodaFi/mystery-flavored-cereal

Tighten-up a Foundation Deserialization Test

swift-ci committed:
Merge pull request #30010 from sjavora/ascii-only-paths-note

swift-ci committed:
Merge pull request #29752 from roop/dev_tips_mem_usage

swift-ci committed:
Merge pull request #31008 from CodaFi/i-just-want-to-be-clean

eeckstein committed:
MergeFunctions: use new LLVM API to create a call.

rdar://problem/61734317

davezarzycki committed:
[line-directive] Escape literal '['

New versions of Python warn if the literal is not escaped. Specifically:

    "Support for nested sets and set operations in regular expressions as in
    Unicode Technical Standard #18 might be added in the future. This would
    change the syntax. To facilitate this future change a FutureWarning will
    be raised in ambiguous cases for the time being. That include sets
    starting with a literal '[' or containing literal character sequences
    '--', '&&', '~~', and '||'. To avoid a warning, escape them with a
    backslash. (Contributed by Serhiy Storchaka in bpo-30349.)"

https://docs.python.org/dev/whatsnew/3.7.html

swift-ci committed:
Merge pull request #31009 from eeckstein/merge-funcs

swift-ci committed:
Merge pull request #31010 from davezarzycki/pr31010

compnerd committed:
Merge pull request #31006 from compnerd/explicit-paths

build: explicitly pass the path to python2,python3

bitjammer committed:
Merge pull request #31000 from bitjammer/acgarland/rdar-61178480-indirect-default-impls

[SymbolGraph] Look at inherited protocols for default implementations

benlangmuir committed:
Merge pull request #30963 from benlangmuir/isdb-san

[build-script] Add sanitizer support to indexstore-db product

shahmishal committed:
Disable creating swift-api-digester dSYM to reduce toolchain size

rintaro committed:
[SourceKit] Don't use clang build sessions when validation is disabled

Don't use '-fbuild-session-timestamp' if
'-disable-modules-validate-system-headers' is specified.

Since system header validation happens under *OR* condition of
'ValidateSystemInputs' and 'ModulesValidateOncePerBuildSession', using
build sessions effectively overrides it.

rdar://problem/61075677

gottesmm committed:
Merge pull request #30942 from gottesmm/debug_compiler_autobisect

[debugging-the-compiler] Add to bisecting optimization passes a short example of how to use llvm's bisect tool to autobisect the optimizer.

gottesmm committed:
[docs] Change two command lines to be an inline literal instead of normal text.

hborla committed:
Merge pull request #30807 from hborla/property-wrapper-refactoring

[Property Wrappers] Refactor property wrappers so the synthesized backing init is only type checked once

gottesmm committed:
[doc] Describe how to use git-bisect in the presence of branch forwarding/feature branches.

shahmishal committed:
Add a comment for removing debug symbols for swift-api-digester from the toolchain

compnerd committed:
runtime: extend shims to x86

The android x86 target exposed the fact that the FP16 support routines
were not being emitted for x86, only x86_64.

compnerd committed:
build: add a workaround for the build to use Python2 instead of Python3

Python3 sometimes doesn't get found, add a temporary workaround.

xedin committed:
[Diagnostics] Allow invalid initializer ref diagnostics point to `.init`

compnerd committed:
Merge pull request #31011 from RLovelett/python3-fix-print

test: Add parens to print for Python3 compatibility

swift-ci committed:
Merge pull request #31017 from gottesmm/pr-891666424206195516c645418a458613034c1d6c

gottesmm committed:
[docs] Add a section on running @swift-ci based non-executable device tests to ContinuousIntegration.md.

dan-zheng committed:
[AutoDiff] Add `REQUIRES: CPU=x86_64` to IRGen test.

xedin committed:
[Diagnostic] Non-ephemeral pointer init diagnostic should point to `.init` if present

shahmishal committed:
Merge pull request #31021 from gottesmm/pr-c115ac2d5c3c0400c2b76de571e9903b70c2adf1

[docs] Add a section on running @swift-ci based non-executable device tests to ContinuousIntegration.md.

rintaro committed:
Merge pull request #31013 from rintaro/sourcekit-buildsession-implicit-rdar61075677

[SourceKit] Don't use clang build sessions when validation is disabled

shahmishal committed:
Merge pull request #31018 from gottesmm/pr-d6426d88ab60dd2b653c227ec20517db745d6131

[doc] Describe how to use git-bisect in the presence of branch forwarding/feature branches.

xedin committed:
[Diagnostics] Missing member diagnostic associated with module should point to name

shahmishal committed:
Merge pull request #31020 from compnerd/python-fallback

build: add a workaround for the build to use Python2 instead of Python3

lorentey committed:
Merge pull request #30601 from lorentey/fortify-benchmarks

[benchmark] Package.swift: Don’t trap on stray files under single-source/

valeriyvan committed:
Removes redundant buffer zeroing in Metal.swift by using `init(unsafeUninitializedCapacity:initializingWith:)` (#30133)

* Removes redandant buffer zeroing

* Fixes typo

* Fix compile error

shahmishal committed:
Merge pull request #31022 from apple/autodiff-testing

[AutoDiff] Add `REQUIRES: CPU=x86_64` to IRGen test.

compnerd committed:
Merge pull request #31019 from compnerd/x86-shims

runtime: extend shims to x86

blueeor committed:
[SR-12460] Don't crash when missing a type in a Diagnostic message. (#30979)

* Don't crash when missing a type in a Diagnostic message.
Print an empty string for the missing type.

Resolves SR-12460

* Added new diagnostic message, eg. "operator '==' declared in extension must be 'static'"

CodaFi committed:
Merge pull request #28435 from brettkoonce/stdlib-unittest-cleanup

clean up some stdlib unittest warnings

shahmishal committed:
Merge pull request #31012 from apple/shahmishal/drop-swift-api-digester-dsym

Disable creating swift-api-digester dSYM to reduce toolchain size

xedin committed:
Merge pull request #30967 from xedin/add-loc-to-failure-diagnostic

[Diagnostics] Introduce `getLoc` and `getSourceRange` to failure diagnostic

CodaFi commented:
@swift-ci test

shahmishal committed:
[Preset] Generate line number debug information only to reduce the symbols toolchain size

CodaFi committed:
Merge pull request #30703 from daeken/patch-1

Fixed incorrect comment in SwiftReflectionTest

CodaFi commented:
@swift-ci test Windows

CodaFi commented:
@HassanElDesouky Could you take a look at removing all of the `setInvalid()s` from redeclaration checking? It might just fix the `ErrorType` thing, but it would be a good cleanup to make here because it's really just resetting the interface type and not actually setting a bit anymore.

HassanElDesouky commented:
> @HassanElDesouky Could you take a look at removing all of the `setInvalid()s` from redeclaration checking? It might just fix the `ErrorType` thing, but it would be a good cleanup to make here because it's really just resetting the interface type and not actually setting a bit anymore.

@CodaFi Just so I understand... You mean remove all of `setInvalid()s` **ONLY** in this function?
https://github.com/apple/swift/blob/426583523e063be1ed75027044f56873b7b73183/lib/Sema/TypeCheckDeclPrimary.cpp#L483

Or do you mean remove all of `setInvalid()s` in the file?

CodaFi commented:
Just in this function would be a good start.  Generally, requests that reset the interface type are also candidates for having their `setInvalid` calls removed, but we’ll start small eh?

HassanElDesouky commented:
Yeah, it's better if we start small. I'll go ahead and remove the `setInvalid`.

CodaFi commented:
Just to be clear, I'm going to merge this. Please shoot us another PR.

HassanElDesouky commented:
Thanks for clearing this out. But since it will be a new PR, maybe remove all of `setInvalid()s` in the TypeCheckDeclPrimary file?

Just to be clear about my task... 
1. I will remove all of `setInvalid()s` and won't substitute them with anything.
2. Since it's only a cleanup, I won't need to update/add new tests.
Is that right?

CodaFi committed:
Merge pull request #30024 from Azoy/calling-metadata-accessor-from-swift!

Runtime: Implement wrapper function to call metadata access functions

CodaFi commented:
> I will remove all of setInvalid()s and won't substitute them with anything.

Yep. Should be a nice sea of red.

> Since it's only a cleanup, I won't need to update/add new tests.
Is that right?

You might need to go update some tests if the diagnostics change, but I wouldn't anticipate much beyond that.

HassanElDesouky commented:
Great! I'll make sure to do that once I wake up *later this day* as it's 4:37 am here in Cairo :) 
Thank you <3

CodaFi committed:
Merge pull request #19954 from Lukasa/cb-bench-h2-enumeration

[benchmark] Add HTTP2StateMachine benchmark

HassanElDesouky commented:
@CodaFi  Is the failure related to my modification?

CodaFi commented:
Yeah. It's just a few tests that no longer emit notes

```
swift/test/Sema/enum_conformance_synthesis.swift:64:42: error: expected note not produced
   var hashValue: String { return "" } // expected-note {{previously declared here}}
```

```
19:56:51   // expected-note@-2 {{'$i' previously declared here}}
19:56:51 ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
19:56:51 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/test/decl/var/property_wrappers.swift:841:6: error: expected note not produced
19:56:51   // expected-note@-3 {{'_i' previously declared here}}
19:56:51 ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
19:56:51 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/test/decl/var/property_wrappers.swift:844:6: error: expected error not produced
19:56:51   // expected-error@-2 {{invalid redeclaration of '$i'}}
19:56:51 ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
19:56:51 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/test/decl/var/property_wrappers.swift:845:6: error: expected error not produced
19:56:51   // expected-error@-3 {{invalid redeclaration of '_i'}}
```

Both of those are good to remove.

swift-ci committed:
Merge pull request #30182 from scentini/enum-cxx

shahmishal committed:
Merge pull request #31029 from apple/shahmishal/line-debug-info-only

[Preset] Generate line number debug information only to reduce the sy…

HassanElDesouky committed:
Merge branch 'SR-12248' of https://github.com/HassanElDesouky/swift into SR-12248

HassanElDesouky committed:
[SR-12248] Remove old errors/notes

HassanElDesouky commented:
@CodaFi I'm sorry I don't know how to edit this correctly so, I'll close this PR and open a new one :)
